analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/LoopDetector.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.CFG.Roslyn;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn;
+
+internal class LoopDetector
+{
+    private readonly HashSet<int> loopBlock;
+
+    public LoopDetector(ControlFlowGraph cfg)
+    {
+        loopBlock = DetectLoopBlockOrdinals(cfg).ToHashSet();
```

Comment:
I considered using a primary constructor, but it's weird because it introduces a new field that we do not need and has the same name as the argument in the method.
I'm not a big fan of in-lining assignments with expression bodies. Usually, we avoid using assignments as expressions for good reason. But I have a somewhat low LOGAF about it.

Reply:
I don't care that much either, i'll leave it as optional.